packages/cli/src/checks/utils.ts

```diff
@@ -187,6 +187,27 @@ export function isArrayEqual(arrA: Array<string>, arrB: Array<string>) {
   return true;
 }
 
+export function getClosestAllowedRange(
+  range: string,
+  allowedVersions: string[]
+) {
+  const major = semver.major(range);
+  const allowedVersionsWithSameMajor = allowedVersions.filter(
+    version => semver.major(version) === major
+  );
+  const possibleRanges =
+    allowedVersionsWithSameMajor.length > 0
+      ? allowedVersionsWithSameMajor
+      : allowedVersions;
+  let closestRange = possibleRanges[0];
+  for (let i = 1; i < possibleRanges.length; i++) {
```

Comment:
can probably make this more readable by sorting the possibleRanges something like:
```
possibleRanges.sort((a,b) => semver.gt(a, b))[0]
```

Reply:
for loops with index iterators are generally a red flag for me in 2022 ;)

Reply:
yep, got rid of the for loop